Update fast/scrolling/mac/scrollbars tests to use new UIHelper scrollbar state function

nmoucht opened the pull request:
#### 3a9451dba31381654b6b9ebd65f21d083a36b474
<pre>
Update fast/scrolling/mac/scrollbars tests to use new UIHelper scrollbar state function
<a href="https://bugs.webkit.org/show_bug.cgi?id=254698">https://bugs.webkit.org/show_bug.cgi?id=254698</a>
rdar://107389069

Reviewed by Simon Fraser.

Update the scrollbar tests to use the new infrastructure. When in non UI-side compositing mode
UIHelper.horizontal/verticalScrollbarState will use the old internals function, and when in
UI-side compositing mode, it will use the new UIScriptController function.

* LayoutTests/fast/scrolling/mac/scrollbars/overflow-in-iframe-overlay-scrollbar-hovered.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overflow-in-iframe-overlay-scrollbar-reveal.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-hovered.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-reveal.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overlay-scrollbar-hovered.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overlay-scrollbar-reveal.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overlay-scrollbar-state-expected.txt:
* LayoutTests/fast/scrolling/mac/scrollbars/overlay-scrollbar-state.html:
* LayoutTests/fast/scrolling/mac/scrollbars/scrollbar-state-expected.txt:
* LayoutTests/fast/scrolling/mac/scrollbars/scrollbar-state.html:
* LayoutTests/fast/scrolling/mac/scrollbars/select-overlay-scrollbar-hovered.html:
* LayoutTests/fast/scrolling/mac/scrollbars/select-overlay-scrollbar-reveal.html:
* LayoutTests/fast/scrolling/mac/scrollbars/very-wide-overlay-scrollbar.html:
* LayoutTests/resources/ui-helper.js:
(window.UIHelper.async waitForConditionAsync):
(window.UIHelper.scrollbarState.return.new.Promise.):
(window.UIHelper.scrollbarState.return.new.Promise):
(window.UIHelper.scrollbarState):
(window.UIHelper.verticalScrollbarState):
(window.UIHelper.horizontalScrollbarState):

Canonical link: <a href="https://commits.webkit.org/262561@main">https://commits.webkit.org/262561@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c5230b6cd541f0b304d9d0905fa5d64c2d289640

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1873 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1904 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1967 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2799 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1968 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1980 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1952 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1735 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1892 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1692 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1673 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2644 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1684 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1659 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1719 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1706 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1736 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1546 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1661 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1670 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/460 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1821 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on LayoutTests/resources/ui-helper.js:
```diff
@@ -860,33 +869,33 @@ window.UIHelper = class UIHelper {
     }
 
-    static scrollbarState(scroller, isVertical)
+    static scrollbarState(scroller, isVertical, internalFunctions)
```
Rather than pass `internalFunctions` which is confusing, I think it would cleaner to have the caller pass the scrollingNodeID, which means we only need the `scrollingNodeID` and `isVertical` arguments.

nmoucht replied:
I don't think this is a desirable solution because then we would have to move the isUsingUISideCompositing() logic into each of the tests.

smfr replied:
Can we get `internalFunctions` as `scroller.document.defaultView.internals` ?

nmoucht replied:
Good idea. Fixed.

nmoucht committed:
Update fast/scrolling/mac/scrollbars tests to use new UIHelper scrollbar state function
https://bugs.webkit.org/show_bug.cgi?id=254698
rdar://107389069

Reviewed by Simon Fraser.

Update the scrollbar tests to use the new infrastructure. When in non UI-side compositing mode
UIHelper.horizontal/verticalScrollbarState will use the old internals function, and when in
UI-side compositing mode, it will use the new UIScriptController function.

* LayoutTests/fast/scrolling/mac/scrollbars/overflow-in-iframe-overlay-scrollbar-hovered.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overflow-in-iframe-overlay-scrollbar-reveal.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-hovered.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-reveal.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overlay-scrollbar-hovered.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overlay-scrollbar-reveal.html:
* LayoutTests/fast/scrolling/mac/scrollbars/overlay-scrollbar-state-expected.txt:
* LayoutTests/fast/scrolling/mac/scrollbars/overlay-scrollbar-state.html:
* LayoutTests/fast/scrolling/mac/scrollbars/scrollbar-state-expected.txt:
* LayoutTests/fast/scrolling/mac/scrollbars/scrollbar-state.html:
* LayoutTests/fast/scrolling/mac/scrollbars/select-overlay-scrollbar-hovered.html:
* LayoutTests/fast/scrolling/mac/scrollbars/select-overlay-scrollbar-reveal.html:
* LayoutTests/fast/scrolling/mac/scrollbars/very-wide-overlay-scrollbar.html:
* LayoutTests/resources/ui-helper.js:
(window.UIHelper.async waitForConditionAsync):
(window.UIHelper.scrollbarState.return.new.Promise.):
(window.UIHelper.scrollbarState.return.new.Promise):
(window.UIHelper.scrollbarState):
(window.UIHelper.verticalScrollbarState):
(window.UIHelper.horizontalScrollbarState):

Canonical link: https://commits.webkit.org/262561@main
